cpp/src/arrow/buffer_builder.h

```diff
@@ -350,6 +350,17 @@ class TypedBufferBuilder<bool> {
     bit_length_ += num_elements;
   }
 
+  void UnsafeAppend(const uint8_t* bitmap, int64_t offset, int64_t num_elements) {
```

Comment:
Add a docstring/comment so that the difference with the previous overload is clear?